Docs: Update private data plane docs (#837)
hasura/ddn-docs: docs/deployment/private-ddn/provision-a-data-plane.mdx

---
sidebar_position: 1
sidebar_label: Provision a Data Plane
description: "Learn how to provision a private Data Plane."
keywords:
  - hasura ddn
  - private ddn
  - dedicated vpc
  - enterprise ddn
seoFrontMatterUpdated: false
---

import Thumbnail from "@site/src/components/Thumbnail";

# How to Provision a Data Plane

## Introduction
A Data Plane in Hasura provides dedicated infrastructure for running your GraphQL workloads. This guide walks you through the process of provisioning a new Data Plane using Hasura's DDN VPC.

## Prerequisites

> You must have dedicated data plane contacts. If you don't have any data plane contract, reach out to our sales team sales [here](https://hasura.io/contact-us). 

Reviewer: ```suggestion
You must have dedicated data plane contracts. If you don't have any data plane contract, reach out to our sales team sales [here](https://hasura.io/contact-us). 
```

## Provisioning Process

### Step 1: Initiate Data Plane Provisioning

> Navigate to the `Data Plane Management` section in your Hasura console. 

Reviewer: ```suggestion
Navigate to the `Data Plane Management` section in your Hasura console ([console.hasura.io](https://console.hasura.io)). 
```

<Thumbnail src="/img/data-plane/data-plane-management.png" alt="Data Plane" width="1000px" />

> You'll see a list of all your contracted Data Planes. Click the `Provision` button to start the provisioning process for the data plane you want to provision.

Reviewer: ```suggestion
You'll see a list of all your Data Planes that you can provision. Click the `Provision` button to start the provisioning process for the data plane.
```

<Thumbnail src="/img/data-plane/pending-provision-data-planes.png" alt="Data Plane" width="1000px" />

> ### step 2: Fill Out the VPC Deployment Form

Reviewer 2: ```suggestion
### Step 2: Fill Out the VPC Deployment Form
```

> To start provisioning for a new Data Plane, you'll need to provide the following information in the deployment form:

Reviewer: ```suggestion
Provide the following information in the form:
```

> <Thumbnail src="/img/data-plane/vpc-form.png" alt="Data Plane" width="1000px" />

Reviewer 2: Let's change the domain to `private-ddn.hasura.app`

Reviewer 3: Done!

- **Name:** Enter a descriptive name using only letters and spaces, between 4-30 characters. This will be used to identify your Data Plane.

> - **Domain:** This field is automatically generated based on your Data Plane name. Format for this is `<data-plane-name>.private-ddn-staging.hasura.app`

Reviewer: Why are we using staging in the domain name? It should be prod, right? @Varun-Choudhary

Reviewer 3: Will change and this. Update the image as well

> - **Cloud Provider:** Select your preferred cloud provider. Currently available options: `AWS`, `Google Cloud` and `Microsoft Azure`. For Google Cloud and Microsoft Azure click on `Request` and fill the request form. This will open a support ticket with Hasura. Our team will reach out to you for more information.

Reviewer: ```suggestion
- **Cloud Provider:** Select your preferred cloud provider. Currently available options: `AWS`, `Google Cloud` and `Microsoft Azure`. For Google Cloud and Microsoft Azure click on `Request` and fill the request form. This will open a support ticket with Hasura. Our team will reach out to you with further instructions.
```

> - **VPC CIDR:** A VPC /16 CIDR block assigned to your Data Plane. Default value: 10.0.0.0/16. 

Reviewer 2: Let's add a note that this configuration cannot be changed later.

> - **Service CIDR:** A /16-/20 CIDR for service workloads. Default value: 172.20.0.0/16

Reviewer 2: ```suggestion
- **Service CIDR:** A /16-/20 CIDR block used for Kubernetes service cluster IP addresses in your Data Plane deployment. This CIDR range is used internally by Kubernetes to assign IP addresses to services running in the cluster. Default value: 172.20.0.0/16.
```

Reviewer 2: Let's add a note that this configuration cannot be changed later.

- **Region:** Select the geographical region for your Data Plane deployment.

> - **Zones:** Select exactly the same number of availability zones as mentioned in you data plane contract.

Reviewer 2: ```suggestion
- **Zones:** Select availability zones using the AZ IDs (e.g. use1-az1, use1-az2) for your Data Plane deployment. The number of zones selected must match your data plane contract.
```
